crates/runtime/src/forge_runtime_extension.rs

```diff
@@ -448,6 +439,140 @@ impl<'a> ExtensionLogic
+                        }
+                        Some(1) => {
+                            let signing_key = p256::ecdsa::SigningKey::random(
+                                &mut p256::elliptic_curve::rand_core::OsRng,
+                            );
+                            let verifying_key = signing_key.verifying_key();
+                            (
+                                signing_key.to_bytes(),
+                                verifying_key.to_encoded_point(false).to_bytes(),
+                            )
+                        }
+                        _ => unreachable!("Invalid EllipticCurve variant"),
+                    }
+                };
+
+                Ok(CheatcodeHandlingResult::Handled(vec![
+                    Felt252::from_bytes_be(&signing_key_bytes[16..32]),
+                    Felt252::from_bytes_be(&signing_key_bytes[0..16]),
+                    Felt252::from_bytes_be(&verifying_key_bytes[17..33]),
+                    Felt252::from_bytes_be(&verifying_key_bytes[1..17]),
+                    Felt252::from_bytes_be(&verifying_key_bytes[49..65]),
+                    Felt252::from_bytes_be(&verifying_key_bytes[33..49]),
+                ]))
+            }
+            "ecdsa_sign_message" => {
+                let private_key_low = inputs[0].clone();
+                let private_key_high = inputs[1].clone();
+                let curve = inputs[2].clone().to_u8();
+                let msg_hash_low = inputs[3].clone();
+                let msg_hash_high = inputs[4].clone();
+
+                let private_key = concat_felts(&private_key_low, &private_key_high);
+                let msg_hash = concat_felts(&msg_hash_low, &msg_hash_high);
+
+                let (r_bytes, s_bytes) = {
+                    match curve {
+                        Some(0) => {
+                            let signing_key =
+                                k256::ecdsa::SigningKey::from_slice(&private_key).unwrap();
+
+                            let signature: k256::ecdsa::Signature =
+                                k256::ecdsa::signature::hazmat::PrehashSigner::sign_prehash(
+                                    &signing_key,
+                                    &msg_hash,
+                                )
+                                .unwrap();
+
+                            signature.split_bytes()
+                        }
+                        Some(1) => {
+                            let signing_key =
+                                p256::ecdsa::SigningKey::from_slice(&private_key).unwrap();
+
+                            let signature: p256::ecdsa::Signature =
+                                p256::ecdsa::signature::hazmat::PrehashSigner::sign_prehash(
+                                    &signing_key,
+                                    &msg_hash,
+                                )
+                                .unwrap();
+
+                            signature.split_bytes()
+                        }
```

Comment:
Should some of these operations be `.expect()`'ed or a Result passed to cairo and panic used there?

Reply:
So currently there is no way to make this `sign_prehash` return `Result::Err`, as it will sign any `u256` correctly using the `signing_key` above. What could be bubbled is the error from `SigningKey::from_slice` above.